src/mcts/search.cc

```diff
@@ -513,15 +512,15 @@ SearchWorker::NodeToProcess SearchWorker::PickNodeToExtend() {
   while (true) {
     {
       SharedMutex::Lock lock(search_->nodes_mutex_);
-      // Check whether we are in the leave.
+      // Check for threading collision...
```

Comment:
I agree that my comment is wrong, but I don't understand your comment too.

What this function does it tries to start scrore update (for now that's just incrementing n-in-flight), and returns whether it was successful (false if collision. Collision is if it's a leaf which is already being processed).

Now somehow that information has to be converted into comment..
```cpp
// Try marking node as being updated, return immediately if collision.
```
?

Reply:
Is that not what "threading collision" means? That we're colliding with some other thread already doing the work?

Reply:
I didn't hear that term before, but anyway it's not true that collision can only come from another thread.
More often if comes from previous elements of current thread's batch.